pkg/deploy/util/util.go

```diff
@@ -66,11 +75,26 @@ func SetDeploymentCondition(status *deployapi.DeploymentConfigStatus, condition
 	}
 	newConditions := filterOutCondition(status.Conditions, condition.Type)
 	status.Conditions = append(newConditions, condition)
+	// Update the metrics with new condition
+	metrics.WithLabelValues(ConditionToMetrics(condition)...).Inc()
 }
 
 // RemoveDeploymentCondition removes the deployment condition with the provided type.
-func RemoveDeploymentCondition(status *deployapi.DeploymentConfigStatus, condType deployapi.DeploymentConditionType) {
+func RemoveDeploymentCondition(status *deployapi.DeploymentConfigStatus, condType deployapi.DeploymentConditionType, metrics *prometheus.GaugeVec) {
+	oldConditions := status.Conditions
 	status.Conditions = filterOutCondition(status.Conditions, condType)
+	// Decrease counters for conditions we are going to remove
```

Comment:
Or doing this at a higher level.  Since this is observational from the cache, you might just want to add a ResourceEventHandler on the cache and count those.